pkg/cloud/services/networking/floatingip.go

```diff
@@ -120,10 +120,14 @@ var backoff = wait.Backoff{
 }
 
 func (s *Service) AssociateFloatingIP(eventObject runtime.Object, fp *floatingips.FloatingIP, portID string) error {
-	s.scope.Logger.Info("Associating floating IP", "id", fp.ID, "ip", fp.FloatingIP)
-
-	if fp.PortID == portID {
-		s.scope.Logger.Info("Floating IP already associated:", "id", fp.ID, "ip", fp.FloatingIP)
+	s.scope.Logger.Info("Associating floating IP", "id", fp.ID, "ip", fp.FloatingIP, "portID", portID)
+
+	// 2 possible situation can reach here: the fp.PortID is not empty
+	// assign to portID, so it's already assigned and no need action
+	// the other is fp.PortID assigned to another fixed ip (the other control plane)
+	// so no actions for both situations
+	if len(fp.PortID) != 0 {
```

Comment:
What do you think about doing this check in the [machine reconciler](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/1a1702811745cfbe82642829692ea6b72459838f/controllers/openstackmachine_controller.go) instead, because `networking.Service#AssociateFloatingIP` is also called from other locations where we potentially want to reassign the FIP, even if it is already assigned to another port

Reply:
yes, updated ,thanks~